tests/infra/runner.py

```diff
@@ -280,6 +280,10 @@ def run(self, max_concurrent=None):
         if max_concurrent is None:
             max_concurrent = len(self.threads)
 
+        if max_concurrent == 0:
+            LOG.warning("No tests to run")
+            return
+
```

Comment:
```suggestion
        assert max_concurrent
```

This is always an error, I see no legitimate use for running no tests at the same time. Alternatively, change the condition on :280 to 'if not max_concurrent' so that 0 becomes equivalent to None and means "all of them at once".

Reply:
Good spot! I didn't realise this was still including in this PR.